privacyidea/lib/tokens/radiustoken.py

```diff
@@ -168,24 +352,92 @@ def split_pin_pass(self, passw, user=None, options=None):
 
         return res, pin, otpval
 
+    @log_with(log)
+    @check_token_locked
+    def authenticate(self, passw, user=None, options=None):
+        """
+        do the authentication on base of password / otp and user and
+        options, the request parameters.
+
+        This is only called after it is verified, that the upper level is no challenge-request
+        or challenge-response
+
+        The "options" are read-only in this method. They are not modified here. authenticate
+        is the last method in the loop ``check_token_list``.
+
+        communication with RADIUS server: yes, if is no previous "radius_result"
+            If there is a "radius" result in the options, we do not query the radius server
+        modification of options: options are not modified
```

Comment:
I think they are modified if we query the radius server -- could you add that?

Reply:
done in 20504484